File: examples/test-projects/small-projects/test-project-cpp/unsafe_demo.cpp
#include "unsafe_demo.h"

#include <cstring>
#include <cstdlib>
#include <cstdio>

// Модуль намеренно содержит потенциально опасные конструкции (ПОК)
// разных категорий CWE — для демонстрации сигнатурного анализа.
void runUnsafeDemo(const char* userInput) {
    char buf[16];
    strcpy(buf, userInput);             // CWE-120: небезопасное копирование строки

    char cmd[64];
    sprintf(cmd, "echo %s", buf);       // CWE-120: форматирование в буфер без контроля длины

    system(cmd);                        // CWE-078: запуск команды ОС

    printf(userInput);                  // CWE-134: неконтролируемая форматная строка
}

File: examples/test-projects/small-projects/test-project-js/unsafeDemo.js
/**
 * UnsafeDemo - демонстрация потенциально опасных конструкций (ПОК)
 * Модуль намеренно содержит опасные паттерны разных категорий CWE
 */
'use strict';

const { exec } = require('child_process');

function runUnsafeDemo(userInput) {
    // CWE-078: Внедрение команд ОС
    exec('echo ' + userInput, function execCallback(err, stdout) {
        if (!err) {
            console.log(stdout);
        }
    });

    // CWE-095: Динамическое исполнение кода через eval
    eval('console.log("' + userInput + '")');

    // CWE-095: Динамическое создание функции
    const fn = new Function('x', 'return ' + userInput);
    fn(userInput);

    // CWE-079: Небезопасная вставка HTML (имитация DOM)
    const element = { innerHTML: '' };
    element.innerHTML = userInput;
}

module.exports = { runUnsafeDemo };
